### PR TITLE
Refactor(repo): Use sharable typescript-config-spirit in tsconfig files

### PR DESCRIPTION
**Pull request by literat**
## Description

Quick refactor of the same settings used across the `tsconfig.json` files.
I am moving them into the common configuration.

### Additional context

https://www.totaltypescript.com/tsconfig-cheat-sheet

### Issue reference

- https://jira.almacareer.tech/browse/DS-1592

### CODE REVIEW

File: exporters/js/tsconfig.json [hunk 1]
```diff
@@ -1,30 +1,10 @@
 {
-  "compileOnSave": true,
+  "extends": "../../configs/typescript-config-spirit/base",
   "compilerOptions": {
-    "baseUrl": ".",
-    "outDir": "./dist",
-    "sourceMap": true,
-    "declaration": true,
-    "moduleResolution": "node",
     "emitDecoratorMetadata": true,
```
Comment (literat) [inline exporters/js/tsconfig.json:4]: **todo (blocking):** We can remove this

Comment (literat) [reply exporters/js/tsconfig.json]: Removed

File: exporters/js/tsconfig.json [hunk 2]
```diff
@@ -1,30 +1,10 @@
...
-    "allowSyntheticDefaultImports": true,
-    "alwaysStrict": true,
-    "forceConsistentCasingInFileNames": true,
-    "noUnusedParameters": false,
-    "noUnusedLocals": true,
-    "strictFunctionTypes": true,
-    "noImplicitAny": true,
-    "esModuleInterop": true,
-    "typeRoots": ["../../node_modules/@types"],
     "lib": ["es2015", "dom", "dom.iterable"],
```
[the remark this answers is not in the document]
Comment (literat) [reply exporters/js/tsconfig.json]: Created.

File: configs/typescript-config-spirit/base.json [hunk 3]
```diff
@@ -0,0 +1,148 @@
+{
+  "compileOnSave": true,
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig to read more about this file */
```
[the remark this answers is not in the document]
Comment (literat) [reply configs/typescript-config-spirit/base.json:4]: Yeah, that is just fine. Associate JSONC with JSON files in your IDE :-)

File: configs/typescript-config-spirit/package.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "typescript-config-spirit",
+  "description": "TypeScript configuration for Spirit Design System",
+  "version": "0.0.0",
```
Comment (adamkudrna) [inline configs/typescript-config-spirit/package.json:4]: v0.1.0 (the lowest correct semver version AFAIK) will be released automatically?

File: configs/typescript-config-spirit/README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+This configuration is the base TypeScript configuration.
+It is fine to use this configuration when you need to transpile Node.js code.
+
+### DOM Configuration
+
+> `typescript-config-spirit/dom`
+
+This configuration extends the base configuration.
+It is fine to use this configuration for the code that is running in the DOM.
```
Comment (adamkudrna) [inline configs/typescript-config-spirit/README.md:44]: Nit-picking: can code be "run in DOM"? 🤔 Didn't you mean "in browser"?

**Review by adamkudrna — COMMENTED** [review]
Looks good, I just don't feel qualified for a ✅.

File: configs/typescript-config-spirit/README.md (hunk 5 above)
Comment (adamkudrna) [reply configs/typescript-config-spirit/README.md:44]: That might imply renaming this config to `browser`. But I expect you will explain why I'm wrong 🙂.

File: configs/typescript-config-spirit/base.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (literat) [reply configs/typescript-config-spirit/base.json:4]: I am not kidding. The JSON format is really bad for storing configurations. Especially with the lack of comments. For a better understanding of configs, comments like these are necessary.

File: configs/typescript-config-spirit/package.json (hunk 4 above)
Comment (literat) [reply configs/typescript-config-spirit/package.json:4]: Yeah, will be re-versioned with the next publish. However, we can start even with the `1.0.0`.

**Review by pavelklibani — APPROVED** [review]

**Review by curdaj — APPROVED** [review]

**Pull request merged**